_posts/2020-07-13-newpipe-0.21.6-released.md

```diff
@@ -0,0 +1,105 @@
+
+[@Stypox](https://github.com/Stypox) slightly improved the layout of streams in the Play Queue. #6443
+
+### Fixed
+
+[@B0pol](https://github.com/B0pol) fixed the extraction of some age-restricted videos on YouTube.
+
+[@evermind-zz](https://github.com/evermind-zz) plugged a memory leak by dismissing any user choice dialogues before exiting an activity.
+
+[@Redirion](https://github.com/Redirion) fixed a crash that occurred when NewPipe connected to certain Bluetooth devices, such as car audio systems.
+
+## Translation
+
+[@chr56](https://github.com/chr56) fixed Simplified Chinese plurals in the app. Thanks!
+
+Also, a huge thanks to all the other translators who keep the many languages of this app up to date every release.
+
+## Nerd Talk
+
+[@TobiGr](https://github.com/TobiGr) added the `streamPosition` attribute for comments. This isn't meant for NewPipe, but for other users of the NewPipe Extractor library who might wish to implement timestamped comments for SoundCloud.
+
+[@FireMasterK](https://github.com/FireMasterK) added `dependabot` to the Extractor repo, which is a GitHub bot that automatically opens PRs to update dependencies one by one, saving the maintainers some manual work. It has already proven useful by updating a few dependencies:
+
+- Bump gson from 2.8.6 to 2.8.7 
+- Bump rhino from 1.7.12 to 1.7.13 
+- Bump spotbugs-annotations from 4.0.2 to 4.2.3
+
+[@TobiGr](https://github.com/TobiGr) updated the hardcoded client ID for SoundCloud.
+
+[@XiangRongLin](https://github.com/XiangRongLin) removed personal IPv4 addresses from the automatically generated mock tests and ensured that they won't slip in again.
+
+Since NewPipe uses Java 8 now, [@Isira-Seneviratne](https://github.com/Isira-Seneviratne) went ahead and converted the abstract class DAOs (data access objects) to interfaces (Read: coder did some coding magic, now things are better somehow, don't ask us why).
+
+[@Imericxu](https://github.com/Imericxu) improved the app code by resolving some Checkstyle warnings related to Tabs (Trending, Subscriptions, etc.).
+
+[@Isira-Seneviratne](https://github.com/Isira-Seneviratne) updated the AndroidX Room library to v2.3.0. This also allowed him to migrate Converters (utility functions which provide easy conversion from one data type to another) to Kotlin.
+
+[@danielmbutler](https://github.com/danielmbutler) replaced the `System.exit` function calls used to close NewPipe with `getActivity.finishAffinity()`. Now instead of straight up killing the app process in one go, the OS will let the app close gracefully by letting it call all its functions to stop activities in the correct order, and free up memory gradually. This improvement was suggested by [@Redirion](https://github.com/Redirion).
```

Comment:
```suggestion
[@danielmbutler](https://github.com/danielmbutler) replaced the `System.exit` function calls used to close NewPipe with `getActivity.finishAffinity()`. Now instead of straight up killing the app process in one go, the OS will let the app close gracefully by allowing it to call all its functions to stop activities in the correct order, and free up memory gradually. This improvement was suggested by [@Redirion](https://github.com/Redirion).
```